# Creating 8 uM solutions from the lyophilized DNA

In [ ]:
# We will start with 4 nmoles of lyophilized DNA and end up with a 8 umol/L (8 uM) solution
print("Add 100 uL of DI water to the lyophilized DNA.")
print("Take 1 uL of the DNA solution, and add 999 uL of DI water to it.")
print("Take 2 uL of the new DNA solution, and add 998 uL of DI water to it.")

# Creating the Initial DNA dilution

In [14]:
print("How many ultramers are being annealed for this part?")
ultramer_number = int(input())
# volumes are in microliters
microliters_water = 983 - 5 * ultramer_number
microliters_3M_salt = 17

print("Add " + str(microliters_water) + " microliters of DI water, 17 microliters of 3M NaCl solution, and 5 microliters" +
      " of each of the " + str(ultramer_number) + " ultramer solutions.")

How many ultramers are being annealed for this part?
2
Add 973 microliters of DI water, 17 microliters of 3M NaCl solution, and 5 microliters of each of the 2 ultramer solutions.


# Amount of DNA in this initial dilution

In [16]:
# amount is in fmol
fmol_DNA = 40000
# concentration is in fmol per uL
concentration = 40000/1000

print("Amount of DNA: " + str(fmol_DNA) + " fmols")
print("Concentration: " + str(concentration) + " fmols/uL")

Amount of DNA: 40000 fmols
Concentration: 40.0 fmols/uL


# Amount of Reagent in the 10 uL final volume

In [19]:
# volumes are in uL
restriction_enzyme = 0.5

# taking 1.75 uL of DNA ligase solution and adding 68.25 uL of water, and we'll be taking volumes from this diluted solution
# this dilution of DNA ligase should have 10 U per uL
ligase = 1

# ligase buffer
ligase_buffer = 1

total_reaction_volume = restriction_enzyme + ligase + ligase_buffer

2.5


# Diluting the DNA solution such that 0.5 uL has 7 femtomoles

In [21]:
# current concentration is 40 fmol/uL
# we want to get this to 14 fmol/uL

print("Add 1.857 mL of DI water to the DNA solutions")

Add 1.857 mL of DI water to the DNA solutions


# Creating a library DNA solution

In [ ]:
# currently we have 2.857 mL of 14 fmol/uL DNA solution
# we'll be left with a 14 fmol/uL DNA solution (where the moles are DNA strand of different parts), with the volume
# depending on how many DNA parts are in the library

print("Add 0.1 mL of each DNA solution into a new library solution")

# Adding DNA to the reaction solution from the 14 fmol/uL solution

In [22]:
print("Add 0.5 uL of DNA to the reaction solution and let DI water fill the remaining solution to 10 uL total")

Add 0.5 uL of DNA to the reaction solution and let DI water fill the remaining solution to 10 uL total


# Protocol for Creating the Working DNA concentrations

##### Add 100 uL of DI water to lyophilized DNA. (40 umol/L)
### Add 20 uL of DNA solution to 80 uL DI water. (8 umol/L)
### Add 973 uL DI water, 17 uL 3M NaCl solution, and 5 uL each of forward/reverse primer. (40 fmol/uL)
### Annealing procedure
Place in 94 C heat block for 3 minutes then allow to return to room temperature on the bench.
### Add 100 uL DNA solution and 185.7 uL DI water. (14 fmol/uL)
### (For library part): Add 50 uL of each DNA solution into a new library solution
** This whole procedure has to be done for every ultramer pairing (which is more than one for sequences longer than 200 bp). **

# Protocol for making the master mix

### Add 35 uL of the restriction enzyme (BbsI)
### Add 1.75 uL of DNA ligase
### Add 70 uL of T4 buffer

# Protocol for filling wells for library assembly

### Add 0.5 uL of DNA solution (14 fmol/uL) for every DNA part.
### Add 1.53 uL of the master mix.
### Add the remaining DI water until reaction volume is 10 uL.
### Throw in thermocycler with the following program:
20 cycles of the following:

37 C for 1.5 minutes
16 C for 3 minutes

50 C for 5 minutes
80 C for 10 minutes


4 C on hold

# Big questions remaining

### How do we treat / prepare our destination vectors?
### How are we going to know which part is which at the beginning of the day?
